plugin/evm/customheader/gas_limit.go

```diff
@@ -188,7 +209,12 @@ func RemainingAtomicGasCapacity(
 		return ap5.AtomicGasLimit, nil
 	}
 
-	state, err := feeStateBeforeBlock(config, parent, header.Time)
+	timeMS := header.Time * 1000
+	if config.IsGranite(header.Time) {
+		timeMS = *customtypes.GetHeaderExtra(header).TimeMilliseconds
+	}
```

Comment:
Check #1288 (comment)

```suggestion
    timeMS := customtypes.HeaderTimeMilliseconds(header)
```